stan/math/prim/fun/lub_constrain.hpp

```diff
@@ -130,18 +114,10 @@ inline return_type_t<T, L, U> lub_constrain(const T& x, const L& lb,
     T exp_minus_x = exp(-x);
     inv_logit_x = inv_logit(x);
     lp += log(ub - lb) - x - 2 * log1p(exp_minus_x);
-    // Prevent x from reaching one unless it really really should.
-    if (x < INFTY && inv_logit_x == 1) {
-      inv_logit_x = 1 - 1e-15;
-    }
   } else {
     T exp_x = exp(x);
     inv_logit_x = inv_logit(x);
     lp += log(ub - lb) + x - 2 * log1p(exp_x);
-    // Prevent x from reaching zero unless it really really should.
-    if (x > NEGATIVE_INFTY && inv_logit_x == 0) {
-      inv_logit_x = 1e-15;
-    }
   }
   return fma(ub - lb, inv_logit_x, lb);
```

Comment:
I know this is preexisting, but the computation of `inv_logit(x)` could be done in just one place (such as in the return line) rather that in both branches of the `if`.

Reply:
It needs to be pulled up above the conditional because the value is used in the condition.

Reply:
Ok if we implement @bob-carpenter 's suggestion.